thewalrus/decompositions.py

```diff
@@ -70,36 +70,33 @@ def williamson(V, rtol=1e-05, atol=1e-08):
     omega = sympmat(n)
     vals = np.linalg.eigvalsh(V)
 
-    for val in vals:
-        if val <= 0:
-            raise ValueError("Input matrix is not positive definite")
+    if not np.all(vals > 0):
+        raise ValueError("Input matrix is not positive definite")
 
     Mm12 = sqrtm(np.linalg.inv(V)).real
     r1 = Mm12 @ omega @ Mm12
     s1, K = schur(r1)
-    X = np.array([[0, 1], [1, 0]])
-    I = np.identity(2)
-    seq = []
-
-    # In what follows I construct a permutation matrix p  so that the Schur matrix has
+    # In what follows a permutation matrix perm1 is constructed so that the Schur matrix has
     # only positive elements above the diagonal
-    # Also the Schur matrix uses the x_1,p_1, ..., x_n,p_n  ordering thus I permute using perm
+    # Also the Schur matrix uses the x_1,p_1, ..., x_n,p_n  ordering thus a permutation perm2 is used
     # to go to the ordering x_1, ..., x_n, p_1, ... , p_n
-
+    perm1 = np.array(range(2 * n))
```

Comment:
can use numpy directly here
```suggestion
    perm1 = np.arange(2 * n)
```